presto-main/src/test/java/com/facebook/presto/block/AbstractTestBlock.java

```diff
@@ -49,10 +50,15 @@
 @Test
 public abstract class AbstractTestBlock
 {
-    protected <T> void assertBlock(Block block, T[] expectedValues)
+    protected <T> void assertBlock(Block block, BlockBuilder blockBuilderTemplate, T[] expectedValues)
     {
-        assertBlockPositions(block, expectedValues);
-        assertBlockPositions(copyBlock(block), expectedValues);
+        assertBlockPositions(block, blockBuilderTemplate, expectedValues);
+        assertBlockPositions(copyBlockViaBlockSerde(block), blockBuilderTemplate, expectedValues);
+        if (expectedValues.getClass().getComponentType().isArray() ||
```

Comment:
Do we ever use array as the expected value? That shouldn't happen because `getObjectValue` doesn't return arrays for any type, as far as I know.

Reply:
@haozhun : Yes. See `TestArrayBlock.testWithArrayBlock`.